fossology/fossology
# feat(rest): Add POST/PATCH license endpoints

## Description

Added new POST and PATCH endpoints to `/license` for creating and editing licenses from REST API.

The GET to `/license` is also edited. Now, the request without license name will list every license (paginated). Request to `/license/<shortname>` will list just the requested license but will also include the obligations associated to it.

### Changes

1. Added the POST and PATCH endpoints to `/license`.
2. New model `Obligation`.
3. Added `/license` endpoint for listing all licenses.
4. Update `/license/<shortname>` endpoint to list license obligations.

## How to test

1. Try to get all the licenses from the `/license` endpoint and check if they matched the DB.
2. Check the pagination feature of the `/license` endpoint.
3. Check the obligation list of a license with `/license/<shortname>` endpoint.
4. Try creating a new license with `POST` request.
5. Try updating a license with `PATCH` request.
